packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,221 @@
+
+const splitDateAndTimeRegex = new RegExp(/^(.*),(.*)$/);
+const dateRegex = new RegExp(/^\s*([0-9]{1,2})\/([0-9]{1,2})\/([0-9]{4})\s*$/);
+const timeRegex = new RegExp(/^\s*([0-9]{1,2}):([0-9]{2})\s*$/);
+
+function hoursAndMinutesToTime(hours, minutes) {
+	return hours * 60 + minutes;
+}
+
+function getTextDate(date, time) {
+	if (date === undefined) {
+		return '';
+	}
+
+	const day = getDate(date);
+	const month = getMonth(date) + 1;
+	const year = getYear(date);
+
+	const dateText = `${twoDigits(day)}/${twoDigits(month)}/${year}`;
+
+	if (time === undefined) {
+		return dateText;
+	}
+
+	const hours = Math.floor(time / 60);
+	const minutes = time % 60;
+
+	const timeText = `, ${twoDigits(hours)}:${twoDigits(minutes)}`;
+
+	return dateText + timeText;
+}
+
+
+class InputDateTimePicker extends React.Component {
+	static propTypes = {
+		selectedDateTime: PropTypes.instanceOf(Date),
+		onChange: PropTypes.func.isRequired,
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = (() => {
+			const selectedDateTime = this.props.selectedDateTime;
+			if (selectedDateTime !== undefined) {
+				const date = selectedDateTime;
+				const hours = getHours(selectedDateTime);
+				const minutes = getMinutes(selectedDateTime);
+				const time = hoursAndMinutesToTime(hours, minutes);
+				return {
+					date,
+					time,
+					textInput: getTextDate(date, time),
+				};
+			}
+
+			return {
+				textInput: '',
+			};
+		})();
+
+		this.onChangeInput = this.onChangeInput.bind(this);
+		this.onSubmitPicker = this.onSubmitPicker.bind(this);
+	}
+
+	onSubmitPicker({ date, time }) {
+		this.updateDateTime(date, time);
+	}
+
+	onChangeInput(event) {
+		const text = event.target.value;
+
+		const splitMatches = text.match(splitDateAndTimeRegex);
+
+		if (!splitMatches) {
+			console.warn('DATETIME - INCORRECT FORMAT:', text);
+			this.updateDateTime(undefined, undefined, text);
+			return;
+		}
+
+		const [
+			,
+			dateText,
+			timeText,
+		] = splitMatches;
+
+		const date = (() => {
+			const dateMatches = dateText.match(dateRegex);
+			if (!dateMatches) {
+				console.warn('DATE - INCORRECT FORMAT:', dateText);
+				return undefined;
+			}
+
+			const [
+				,
+				dayString,
+				monthString,
+				yearString,
+			] = dateMatches;
+
+			const day = parseInt(dayString, 10);
+			const month = parseInt(monthString, 10);
+			const monthIndex = month - 1;
+			const year = parseInt(yearString, 10);
+
+			if (month === 0 || month > 12) {
+				console.warn('DATE - INCORRECT MONTH NUMBER:', dateText);
+				return undefined;
+			}
+
+			if (day === 0) {
+				console.warn('DATE - INCORRECT DAY NUMBER:', dateText);
+				return undefined;
+			}
+
+			const monthDate = new Date(year, monthIndex);
+			const lastDateOfMonth = lastDayOfMonth(monthDate);
+
+			if (day > getDate(lastDateOfMonth)) {
+				console.warn('DATE - INCORRECT DAY NUMBER RELATIVE TO MONTH:', dateText);
+				return undefined;
+			}
+
+			const dateValidated = setDate(monthDate, day);
```

Comment:
Instead of performing all the checks yourself, is there a way with Date or date-fns to try to create a date from `dateText`, and get undefined if it's not valid ?
The less we write code, the better, and this seems to be something quite basic in date manipulation libs.

Reply:
I didn't think to much of it, I confess.
I was mostly thinking, this part will need to be rework because of the internationalization thingy.
And have a parser which take every custom local format is not a common thing (not impossible though).
The `date-fns` parser take only exact ISO string format not visual custom one like ours. And also, like we did it for user experience, we need to extract date or time only when valid before the full date is.

I was thinking, when we'll do all the configuration format, to extract the input part into a specific component and deal with a lib or a bunch of custom parsers/formatters.

Reply:
But if you find something that takes the string and the `format` it could be good. And we don't necessarily need to know what part precisely is wrong, but telling the user, the input doesn't follow the wanted format (and giving him the format) is totally fine.

Reply:
We need to check specifically the parts because of relation between day and month/year (29/02 could be valid or not, depending on year ; 31 is valid for some month but not all, etc.).
How to you constrain format and allow '14/01/2018', '01/01/2018', '10/01/2018' but not '00/01/2018' ?
I don't know you can do it with a simple regexp and just anyway with a regexp.
The same for month before 1 and after 12.

After that, like I did, we extract the date or the time even if the entire string don't match the format. Allowing a quick real time feedback in the picker.

We'll see to update/rework that an other time, with l10n.
✘